client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -93,24 +93,32 @@
         private long eventNumber = 0;
         private final ReusableLatch connectionSetup = new ReusableLatch();
         @GuardedBy("lock")
-        private CompletableFuture<Void> emptyInflightFuture = null;
+        private CompletableFuture<Exception> emptyInflightFuture = null;
 
         /**
          * Returns a future that will complete successfully once all the inflight events are acked
          * and will fail if that is not possible for some reason. IE: because the connection
          * dropped, or the segment was sealed.
          */
-        private CompletableFuture<Void> getEmptyInflightFuture() {
+        private CompletableFuture<Exception> getEmptyInflightFuture() {
```

Comment:
Actually, looking into this more, I think a utility method for futures would help with this situation a lot. I'll add it and update the PR.